src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/ConvertToRecordHelpers.cs

```diff
@@ -278,6 +335,47 @@ public static bool IsDefaultEqualsOperator(IMethodBodyOperation operation)
             return default;
         });
 
+        /// <summary>
+        /// Determines whether the operation is safe to move into the "this(...)" initializer
+        /// i.e. Doesn't reference any other created variables but the parameters
+        /// </summary>
+        private static bool IsSafeAssignment(IOperation operation)
+            => operation.WalkDownConversion() switch
+            {
+                // literals don't make references
+                ILiteralOperation or
+                IParameterReferenceOperation or
+                IDefaultValueOperation => true,
+                IMemberReferenceOperation memberReference
+                    => memberReference.Instance == null ||
+                    IsSafeAssignment(memberReference.Instance),
+                IInvocationOperation invocation
+                    => (invocation.Instance == null ||
+                        IsSafeAssignment(invocation.Instance)) &&
+                    invocation.Arguments.All(arg => IsSafeAssignment(arg.Value)),
+                IUnaryOperation unary => IsSafeAssignment(unary.Operand),
```

Comment:
Can we have a generic catch-all that just walks the child operations without having to special case all of these?